### PR TITLE
Added support for clientOptions in generated code; Fixed several ApiView issues 

### PR DESCRIPTION
**Pull request by LarryOsterman**
# Added support for clientOptions in generated code; removed a few sync withResponse APIs that were accidentally left in; Removed a bunch of unused generated code; added tests for telemetry and distributed tracing

Code updates include:
 - Reduced the generated API surface by removing sync and other unused generated APIs.
 - Added `clientOptions` to builder classes.
 - Removed a few synchronous attestation APIs that shouldn't have been created.
 - Stopped using the auto-generated builder in favor of manually creating the pipeline to ensure that User-Agent is correctly set.
 - Fixed a couple of telemetry related bugs and added tests to verify that tracing headers are correctly generated.

# All SDK Contribution checklist:
- [X] **The pull request does not introduce [breaking changes]**
- [X] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](../../CONTRIBUTING.md#building-and-unit-testing)
- [X] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationAsyncClient.java [hunk 1]
```diff
@@ -290,12 +290,12 @@
...
      * @return cached signers.
      */
-    Mono<List<AttestationSigner>> getCachedAttestationSigners() {
+    Mono<List<AttestationSigner>> getCachedAttestationSigners(Context context) {
         if (this.cachedSigners.get() != null) {
             return Mono.just(this.cachedSigners.get());
         } else {
-            return this.signerImpl.getAsync()
-                .map(AttestationSignerImpl::attestationSignersFromJwks)
+            return this.signerImpl.getWithResponseAsync(context)
```
Comment (pallavit) [inline sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationAsyncClient.java:297]: To make sure I understand this we added context through all the Rest API calls for open telemetry to work correctly.

File: sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationClientBuilder.java [hunk 2]
```diff
@@ -233,53 +331,75 @@ public AttestationClientBuilder tokenValidationOptions(AttestationTokenValidatio
...
+            ? serviceVersion
+            : AttestationServiceVersion.getLatest();
+
+        // endpoint cannot be null, which is required in request authentication
+        String endpoint = this.endpoint;
+        Objects.requireNonNull(endpoint, "'Endpoint' is required and can not be null.");
+
+        // If the customer provided a pipeline, use it, otherwise configure the pipeline based on the options
+        // which were provided.
+        HttpPipeline pipeline = this.pipeline;
```
Comment (pallavit) [inline sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationClientBuilder.java:358]: Other than the user-agent bug what else motivated you to create your own httppipeline here?

Comment (pallavit) [reply sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationClientBuilder.java:358]: I will make sure we capture those to make our inner builders better.

File: sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/implementation/AttestationClientImpl.java [hunk 3]
```diff
@@ -128,7 +128,7 @@ public MetadataConfigurationsImpl getMetadataConfigurations() {
      * @param instanceUrl The attestation instance base URI, for example https://mytenant.attest.azure.net.
      * @param apiVersion Api Version.
      */
-    AttestationClientImpl(String instanceUrl, String apiVersion) {
+    public AttestationClientImpl(String instanceUrl, String apiVersion) {
```
Comment (pallavit) [inline sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/implementation/AttestationClientImpl.java:131]: Did you hand edit this?

Comment (pallavit) [reply sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/implementation/AttestationClientImpl.java:131]: Just saw the autorest changes.

File: sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationAsyncClient.java (hunk 1 above)
Comment (LarryOsterman) [reply sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationAsyncClient.java:297]: Almost exactly correct. When I removed the sync stubs, the getAsync method was removed and I changed the code to use getWithResponseAsync. But I originally screwed up and didn't plumb context through, so while working through the telemetry changes, I fixed that.

File: sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationClientBuilder.java (hunk 2 above)
Comment (LarryOsterman) [reply sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationClientBuilder.java:358]: It was entirely the user-agent bug. I was also following the examples used by all the other Java SDKs that I looked at.

**Review by pallavit — APPROVED** [review]
:shipit:

**Pull request merged**